src/server/qgsserverprojectutils.h

```diff
@@ -320,16 +322,18 @@ namespace QgsServerProjectUtils
   SERVER_EXPORT QStringList wmsRestrictedComposers( const QgsProject &project );
 
   /**
-   * Returns the WMS service url defined in a QGIS project.
+   * Returns the WMS service url.
```

Comment:
I believe we need a much better explanation about the internal logic: from where the values are taken and in which order.

Reply:
Can you please make sure there are tests for this method? (I didn't check if you already have).

Reply:
Yes, I already do a test case :-)